### PR TITLE
Rename some builder classes

### PR DESCRIPTION
**Pull request by enbrain**
Cherry-picked class name changes from #1940

### CODE REVIEW

File: mappings/net/minecraft/advancement/Advancement.mapping [hunk 1]
```diff
@@ -30,7 +30,7 @@ CLASS net/minecraft/class_161 net/minecraft/advancement/Advancement
 	METHOD method_690 addChild (Lnet/minecraft/class_161;)V
 		ARG 1 child
 	METHOD method_691 getRewards ()Lnet/minecraft/class_170;
-	CLASS class_162 Task
+	CLASS class_162 Builder
```
Comment (liach) [inline mappings/net/minecraft/advancement/Advancement.mapping:33]: :+1: please roast me

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/data/server/recipe/CookingRecipeJsonBuilder.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_2454 net/minecraft/data/server/recipe/CookingRecipeJsonFactory
+CLASS net/minecraft/class_2454 net/minecraft/data/server/recipe/CookingRecipeJsonBuilder
 	FIELD field_11414 experience F
 	FIELD field_11415 cookingTime I
 	FIELD field_11416 builder Lnet/minecraft/class_161$class_162;
```
Comment (Juuxel) [inline mappings/net/minecraft/data/server/recipe/CookingRecipeJsonBuilder.mapping]: Should this be `advancementBuilder`? (Same in `Shaped/Shapeless/SingleItem/SmithingRecipeJsonBuilder`)

**Review by Juuxel — APPROVED** [review]

**Comment by Juuxel** [comment]
This PR might be impactful if people are already using these classes with the datagen API in Fabric API (Specifically when implementing their [`FabricRecipesProvider`](https://github.com/FabricMC/fabric/blob/1.18/fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricRecipesProvider.java))

**Comment by liach** [comment]
This one is actually impactful for the advancement builder, but we need that rename.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**